imapclient/client_test.go

```diff
@@ -62,6 +63,27 @@ func newClientServerPair(t *testing.T, initialState imap.ConnState) (*imapclient
 	return client, server
 }
 
+// bad server: connected but without greeting
+func newClientWithBadServer(t *testing.T) (*imapclient.Client, io.Closer) {
+	ln, err := net.Listen("tcp", "localhost:10004")
+	if err != nil {
+		t.Fatalf("net.Listen() = %v", err)
+	}
+
+	conn, err := net.Dial("tcp", ln.Addr().String())
+	if err != nil {
+		t.Fatalf("net.Dial() = %v", err)
+	}
+
+	client := imapclient.New(conn, nil)
+
+	go func() {
+		time.AfterFunc(time.Millisecond, func() { ln.Close() })
```

Comment:
No need to wait here, we can close the listener right away since the client already established a connection off of it.

TBH maybe we should be using `net.Pipe` instead?